### ISSUE TITLE
Remove Unnecessary Memory Copies w/go1.20

### ISSUE THREAD

**Issue by patrick-ogrady** [ava-labs/hypersdk#23]
go1.20 adds a new feature (https://github.com/golang/go/issues/46505) that allows us to cast slices to fixed-size arrays.

This will dramatically reduce memory allocations in block processing. Thanks to @rafael-abuawad for calling my attention to this in #21.

I wish we had more memory allocations benchmarked so we could show the performance improvement here but don't think it makes sense to wait to add those to add this.

**Comment by rafael-abuawad** [comment]
#21 A optimization that I found was changing this function:
```go
// PublicKey returns a PublicKey associated with the Ed25519 PrivateKey p.
// The PublicKey is the last 32 bytes of p.
func (p PrivateKey) PublicKey() PublicKey {
	rpk := p[32:] // privateKey == private|public
	var pk PublicKey
	copy(pk[:], rpk)
	return pk
}
```
With this:
```go
// PublicKey returns a PublicKey associated with the Ed25519 PrivateKey p.
// The PublicKey is the last 32 bytes of p.
func (p PrivateKey) PublicKey() PublicKey {
	return PublicKey(p[PrivateKeyLen:])
}
```

**Comment by patrick-ogrady** [comment]
This work is now unblocked!

**Comment by patrick-ogrady** [comment]
This will likely involve converting a bunch of arrays to pointers to arrays so we can reuse memory from parsed blocks.

**Comment by aaronbuchwald** [comment]
Closing this issue - will open a new issue if this shows up in metrics/profiling from new load testing
